### PR TITLE
docs(kic): add compatibilities of router flavors of Kong

### PR DESCRIPTION
**Pull request by randmonkey**
### Summary

Add compatibility table to show which kong router flavors are supported in KIC.

### Reason

fixes [kuberneres-ingress-controller#3187](https://github.com/Kong/kubernetes-ingress-controller/issues/3187)

### Testing

https://deploy-preview-4834--kongdocs.netlify.app/kubernetes-ingress-controller/latest/references/supported-router-flavors/

### Special Notes

Kong 3.0.x `traditional_compatible` router is not fully supported in KIC because `traditional_compatible` router does not support some routes supported in `traditional` router. How should we mark it in the compatibility table?
- use a :heavy_check_mark: and add annotations to tell the incompatibilities
- use a :negative_squared_cross_mark: and add annotations to tell that most cases are supported (now this is used)
- use a symbol other than these two, or tell by text such as `Not Fully Supported`

### CODE REVIEW

File: src/kubernetes-ingress-controller/references/supported-router-flavors.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+title: Supported Kong Router Flavors
+---
+
+## Supported Kong Router Flavors
+
+{{site.base_gateway}} (open-source and enterprise edition) has included the new [expression based router][gateway-expression-router] in version 3.0 and later. 
+The router of {{site.base_gateway}} could be configured to the following [modes][gateway-router-flavor]:
+
+- `traditional`: uses the pre-3.0 router.
+- `traditional_compatible`: uses the expression based router, but the rotuer configure interface remains the same as `traditional`.
```
Comment (lena-larionova) [inline src/kubernetes-ingress-controller/references/supported-router-flavors.md:11]:
```suggestion
- `traditional_compatible`: uses the expression based router, but the router configuration interface remains the same as `traditional`.
```

File: src/kubernetes-ingress-controller/references/supported-router-flavors.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+- `expression`: uses expression based rputer, and expressions must be provided in router configure interface.
```
Comment (lena-larionova) [inline src/kubernetes-ingress-controller/references/supported-router-flavors.md:12]:
```suggestion
- `expression`: uses the expression-based router, and expressions must be provided in the router configuration interface.
```

File: src/kubernetes-ingress-controller/references/supported-router-flavors.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
+---
+title: Supported Kong Router Flavors
+---
+
+## Supported Kong Router Flavors
+
+{{site.base_gateway}} (open-source and enterprise edition) has included the new [expression based router][gateway-expression-router] in version 3.0 and later.
```
Comment (lena-larionova) [inline src/kubernetes-ingress-controller/references/supported-router-flavors.md:7]:
```suggestion
{{site.base_gateway}} (open-source and enterprise edition) includes an [expression-based router][gateway-expression-router] in versions 3.0 and later. 
```

File: src/kubernetes-ingress-controller/references/supported-router-flavors.md [hunk 4]
```diff
@@ -0,0 +1,27 @@
+---
+title: Supported Kong Router Flavors
+---
+
+## Supported Kong Router Flavors
+
+{{site.base_gateway}} (open-source and enterprise edition) has included the new [expression based router][gateway-expression-router] in version 3.0 and later. 
+The router of {{site.base_gateway}} could be configured to the following [modes][gateway-router-flavor]:
```
Comment (lena-larionova) [inline src/kubernetes-ingress-controller/references/supported-router-flavors.md:8]:
```suggestion
The router can be configured in the following [modes][gateway-router-flavor]:
```

File: src/kubernetes-ingress-controller/references/supported-router-flavors.md [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+
+- `traditional`: uses the pre-3.0 router.
+- `traditional_compatible`: uses the expression based router, but the rotuer configure interface remains the same as `traditional`.
+- `expression`: uses expression based rputer, and expressions must be provided in router configure interface.
+
+The compatabilities of router flavors between different {{site.kic_product_name}} version and {{site.base_gateway}} is shown in the following table.
```
Comment (lena-larionova) [inline src/kubernetes-ingress-controller/references/supported-router-flavors.md:14]:
```suggestion
The compatibilities of router flavors between different {{site.kic_product_name}} versions and {{site.base_gateway} are shown in the following table.
```

File: src/kubernetes-ingress-controller/references/supported-router-flavors.md [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+
+| {{site.kic_product_name}}           | 2.7.x                           |
+|:-----------------------------------:|:-------------------------------:|
+| Kong 3.0.x  traditional             |  <i class="fa fa-check"></i>    |
```
Comment (lena-larionova) [inline src/kubernetes-ingress-controller/references/supported-router-flavors.md:18]:
```suggestion
| Kong 3.0.x  `traditional`            |  <i class="fa fa-check"></i>    |     
```

File: src/kubernetes-ingress-controller/references/supported-router-flavors.md [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+- `traditional`: uses the pre-3.0 router.
+- `traditional_compatible`: uses the expression based router, but the rotuer configure interface remains the same as `traditional`.
+- `expression`: uses expression based rputer, and expressions must be provided in router configure interface.
+
+The compatabilities of router flavors between different {{site.kic_product_name}} version and {{site.base_gateway}} is shown in the following table.
+
+| {{site.kic_product_name}}           | 2.7.x                           |
+|:-----------------------------------:|:-------------------------------:|
+| Kong 3.0.x  traditional             |  <i class="fa fa-check"></i>    |     
+| Kong 3.0.x  traditional_compatible  |  <i class="fa fa-check"></i>(*) |
```
Comment (lena-larionova) [inline src/kubernetes-ingress-controller/references/supported-router-flavors.md:19]:
```suggestion
| Kong 3.0.x  `traditional_compatible`  |  <i class="fa fa-check"></i>(*) | 
```

File: src/kubernetes-ingress-controller/references/supported-router-flavors.md [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+- `traditional_compatible`: uses the expression based router, but the rotuer configure interface remains the same as `traditional`.
+- `expression`: uses expression based rputer, and expressions must be provided in router configure interface.
+
+The compatabilities of router flavors between different {{site.kic_product_name}} version and {{site.base_gateway}} is shown in the following table.
+
+| {{site.kic_product_name}}           | 2.7.x                           |
+|:-----------------------------------:|:-------------------------------:|
+| Kong 3.0.x  traditional             |  <i class="fa fa-check"></i>    |     
+| Kong 3.0.x  traditional_compatible  |  <i class="fa fa-check"></i>(*) | 
+| Kong 3.0.x  expression              |  <i class="fa fa-times"></i>    |
```
Comment (lena-larionova) [inline src/kubernetes-ingress-controller/references/supported-router-flavors.md:20]:
```suggestion
| Kong 3.0.x  `expression`              |  <i class="fa fa-times"></i>    |
```

File: src/kubernetes-ingress-controller/references/supported-router-flavors.md [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+
+(*) Most use cases are supported. Regexed with a backslash (`\`) followed by a non-escaped character (for example `\j`) in matches of paths or headers
```
Comment (lena-larionova) [inline src/kubernetes-ingress-controller/references/supported-router-flavors.md:22]:
```suggestion
(*) Most use cases are supported. Regexes with a backslash (`\`) followed by a non-escaped character (for example, `\j`) in matches of paths or headers
```

File: src/kubernetes-ingress-controller/references/supported-router-flavors.md [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+may not be accepted by Kong 3.0 configured to use `traditional_compatible` router.
```
Comment (lena-larionova) [inline src/kubernetes-ingress-controller/references/supported-router-flavors.md:23]:
```suggestion
may not be accepted when {{site.base_gateway}} 3.0 is configured to use the `traditional_compatible` router.
```

**Comment by lena-larionova** [comment]
Ah, review might've been early, I only just saw that this is in draft :) was trying to clear the queue.

File: src/kubernetes-ingress-controller/references/supported-router-flavors.md (hunk 1 above)
Comment (randmonkey) [reply src/kubernetes-ingress-controller/references/supported-router-flavors.md]: fixed

File: src/kubernetes-ingress-controller/references/supported-router-flavors.md (hunk 2 above)
Comment (randmonkey) [reply src/kubernetes-ingress-controller/references/supported-router-flavors.md]: fixed

File: src/kubernetes-ingress-controller/references/supported-router-flavors.md (hunk 3 above)
Comment (randmonkey) [reply src/kubernetes-ingress-controller/references/supported-router-flavors.md]: fixed

File: src/kubernetes-ingress-controller/references/supported-router-flavors.md (hunk 5 above)
Comment (randmonkey) [reply src/kubernetes-ingress-controller/references/supported-router-flavors.md]: fixed

File: src/kubernetes-ingress-controller/references/supported-router-flavors.md (hunk 6 above)
Comment (randmonkey) [reply src/kubernetes-ingress-controller/references/supported-router-flavors.md]: fixed

File: src/kubernetes-ingress-controller/references/supported-router-flavors.md (hunk 8 above)
Comment (randmonkey) [reply src/kubernetes-ingress-controller/references/supported-router-flavors.md]: fixed

File: src/kubernetes-ingress-controller/references/supported-router-flavors.md (hunk 10 above)
Comment (randmonkey) [reply src/kubernetes-ingress-controller/references/supported-router-flavors.md]: fixed

File: src/kubernetes-ingress-controller/references/supported-router-flavors.md (hunk 9 above)
Comment (randmonkey) [reply src/kubernetes-ingress-controller/references/supported-router-flavors.md]: fixed

**Comment by randmonkey** [comment]
> Ah, review might've been early, I only just saw that this is in draft :) was trying to clear the queue.

Thank you. Fixed for all the comments.

File: src/kubernetes-ingress-controller/references/supported-router-flavors.md [hunk 11]
```diff
@@ -0,0 +1,27 @@
+---
+title: Supported Kong Router Flavors
+---
+
+## Supported Kong Router Flavors
```
Comment (lena-larionova) [inline src/kubernetes-ingress-controller/references/supported-router-flavors.md:5]:
```suggestion
```
This title is just a duplicate of the page title, so it's not needed.

**Review by lena-larionova — APPROVED** [review]
LGTM, thanks for the new doc!

**Pull request merged**